pytest/tests/sanity/rpc_finality.py

```diff
@@ -1,66 +1,108 @@
-        # kill one validating node so that no block can be finalized
-        nodes[2].kill()
-        time.sleep(1)
+        consensus = {
+            "consensus": {
+                "min_block_production_delay": {
+                    "secs": min_block_delay,
+                    "nanos": 0,
+                },
+                "max_block_production_delay": {
+                    "secs": min_block_delay * 2,
+                    "nanos": 0,
+                },
+                "max_block_wait_delay": {
+                    "secs": min_block_delay * 3,
+                    "nanos": 0,
+                }
+            }
+        }
+
+        config = {
+            0: {
+                "consensus": consensus
+            },
+            1: {
+                "consensus": consensus
+            },
+            2: {
+                "consensus": consensus
+            },
+        }
+
+        nodes: List[LocalNode] = start_cluster(3, 0, 1, None, [
+            ["min_gas_price", 0],
+            ["epoch_length", 100],
+        ], config)
+
+        utils.wait_for_blocks(nodes[0], target=3)
 
-        acc0_balance = int(nodes[0].get_account('test0')['result']['amount'])
-        acc1_balance = int(nodes[0].get_account('test1')['result']['amount'])
+        balances = {
+            account: int(nodes[0].get_account('test0')['result']['amount'])
+            for account in ['test0', 'test1']
+        }
 
         token_transfer = 10
         latest_block_hash = nodes[0].get_latest_block().hash_bytes
         tx = sign_payment_tx(nodes[0].signer_key, 'test1', token_transfer, 1,
                              latest_block_hash)
         logger.info("About to send payment")
-        logger.info(nodes[0].send_tx_and_wait(tx, timeout=200))
+        # this transaction will be added to the block (probably around block 5)
+        # and the the receipts & transfers will happen in the next block (block 6).
+        # This function should return as soon as block 6 arrives in node0.
+        logger.info(nodes[0].send_tx_and_wait(tx, timeout=10))
         logger.info("Done")
 
-        # wait for doomslug finality
-        time.sleep(5)
-        for i in range(2):
-            acc_id = 'test0' if i == 0 else 'test1'
-            acc_no_finality = nodes[0].get_account(acc_id)
-            acc_doomslug_finality = nodes[0].get_account(acc_id, "near-final")
-            acc_nfg_finality = nodes[0].get_account(acc_id, "final")
-            if i == 0:
-                self.assertEqual(int(acc_no_finality['result']['amount']),
-                                 acc0_balance - token_transfer)
-                self.assertEqual(int(acc_doomslug_finality['result']['amount']),
-                                 acc0_balance - token_transfer)
-                self.assertEqual(int(acc_nfg_finality['result']['amount']),
-                                 acc0_balance - token_transfer)
+        # kill one validating node so that block cannot be finalized.
+        nodes[2].kill()
```

Comment:
Shouldn't we kill the node **before** sending tx to eliminate a chance that it will sign the tested block? (and sending some dummy request to it to check that it timeouts)

Reply:
That's why I increased the block production time by a lot. If we kill the node before sending - then the block will never be signed (which means that the tokens will never be added to the receiver - even in optimistic query)